keystorage/aws/src/test/java/tech/pegasys/signers/aws/AwsSecretsManagerTest.java

```diff
@@ -71,255 +73,287 @@ private void verifyEnvironmentVariables() {
-    secretsManagerClient.close();
-  }
-
-  void createSecret(
-      final boolean multipleSecrets, final boolean multipleTags, final boolean sharedTag) {
-    secretNamePrefix = "signers-aws-integration/";
-    secretName = secretNamePrefix + UUID.randomUUID();
-    secretNames.add(secretName);
-
-    final List<Tag> tags = new ArrayList<>();
-
-    if (sharedTag) {
-      secretTags
-          .entrySet()
-          .forEach(
-              entry -> tags.add(Tag.builder().key(entry.getKey()).value(entry.getValue()).build()));
-    } else if (multipleTags) {
-      tags.add(Tag.builder().key(secretNamePrefix + UUID.randomUUID()).value(secretName).build());
-    }
-
-    tags.add(Tag.builder().key(secretNamePrefix + UUID.randomUUID()).value(secretName).build());
-
-    tags.forEach(
-        tag -> {
-          secretTags.put(tag.key(), tag.value());
-        });
-
-    final CreateSecretRequest secretRequest =
-        CreateSecretRequest.builder()
-            .name(secretName)
-            .secretString(SECRET_VALUE)
-            .tags(tags)
-            .build();
-    secretsManagerClient.createSecret(secretRequest);
-
-    if (multipleSecrets) {
-      createSecret(false, multipleTags, sharedTag);
-    }
-  }
-
-  @AfterEach
-  void deleteSecrets() {
-    secretNames.forEach(
-        name -> {
-          final DeleteSecretRequest secretRequest =
-              DeleteSecretRequest.builder().secretId(name).build();
-          secretsManagerClient.deleteSecret(secretRequest);
-        });
-    secretNames.clear();
-    secretTags.clear();
-  }
-
-  private void validateMappedSecret(
-      final Collection<AbstractMap.SimpleEntry<String, String>> secretEntries,
-      final String secretName) {
-    final Optional<AbstractMap.SimpleEntry<String, String>> secretEntry =
-        secretEntries.stream().filter(e -> e.getKey().equals(secretName)).findAny();
-    assertThat(secretEntry).isPresent();
-    assertThat(secretEntry.get().getValue()).isEqualTo(SECRET_VALUE);
-  }
-
   @BeforeAll
-  void setup() {
-    verifyEnvironmentVariables();
-    setupSecretsManagers();
-    setupSecretsManagerClient();
-    initializeVariables();
+  static void setup() {
+    initAwsSecretsManagers();
+    initTestSecretsManager();
+    initTestVariables();
   }
 
   @AfterAll
-  void teardown() {
+  static void teardown() {
     if (awsSecretsManagerDefault != null
         || awsSecretsManagerExplicit != null
-        || secretsManagerClient != null) {
+        || testSecretsManagerClient != null) {
       deleteSecrets();
       closeClients();
     }
   }
 
+  @BeforeEach
+  void initTestClients() {
+    initAwsSecretsManagers();
+  }
+
+  @AfterEach
+  void closeTestClients() {
+    closeAwsSecretsManagers();
+  }
+
   @Test
   void fetchSecretWithDefaultManager() {
-    createSecret(false, false, false);
-    Optional<String> secret = awsSecretsManagerDefault.fetchSecret(secretName);
+    createSecret(false, false);
```

Comment:
Is there any benefit to creating the secrets for each test? Could we just create them all in a beforeAll so simplify the tests?